cpp/src/arrow/python/datetime.h

```diff
@@ -113,6 +113,43 @@ inline int64_t PyDateTime_to_ns(PyDateTime_DateTime* pydatetime) {
   return PyDateTime_to_us(pydatetime) * 1000;
 }
 
+ARROW_PYTHON_EXPORT
+inline int64_t PyDelta_to_s(PyDateTime_Delta* pytimedelta) {
+  int64_t total_seconds = 0;
+  // TODO(py2) change to "PyDateTime_DELTA_GET_SECONDS/DAYS(pytimedelta)"
+  total_seconds +=
+      PyLong_AsLongLong(PyObject_GetAttrString((PyObject*)pytimedelta, "seconds"));
```

Comment:
The previous version of this code didn't use attribute access. Why did you switch to this? It's gonna be a lot slower.

Reply:
So this is the python 2 issue discussed on chat. I can switch to #if for python 2 / 3

Reply:
As pointed out on chat, you can use C attribute access on py2.